### ISSUE TITLE
Re-include default aspect argument?

### ISSUE THREAD

**Issue by spencerahill** [spencerkclark/facets#38]
Hi @spencerkclark!

I recently created a fresh conda environment for the first time in quite a few months, and evidently faceted has been updated since then w/ what appears to be a breaking change.  `aspect` no longer is set to the golden ratio by default, and instead one must specify at least two of height, width, and aspect.

I know complaining about breaking changes at such an early stage of a package is a bit of a nuisance, but still I'm wondering if it would be worth reintroducing the default `aspect=0.618` setting in the future.  Not just for the sake of my legacy code...a really nice feature of faceted for me was being able to just do `fig, ax = faceted_ax()` or e.g. `fig, axarr = faceted(1, 2)`.  I think that simplicity of call signature is something worth retaining in the future.  Did the generalizing you've done make that no longer appropriate?

I can workaround this obviously with a simple wrapper to faceted and faceted_ax, but wanted to get your take for the sake of faceted more broadly.  Thanks!

**Comment by spencerahill** [comment]
FYI here are the wrappers (included a default width as well):

```python

import faceted.faceted as fac_faceted
import faceted.faceted_ax as fac_ax

def faceted(*args, width=4, aspect=0.618, **kwargs):
    """Wrapper to faceted.faceted w/ a default aspect ratio."""
    return fac_faceted(*args, width=width, aspect=aspect, **kwargs)


def faceted_ax(*args, width=4, aspect=0.618, **kwargs):
    """Wrapper to faceted.faceted_ax w/ a default aspect ratio."""
    return fac_ax(*args, width=width, aspect=aspect, **kwargs)
```

**Comment by spencerkclark** [comment]
All good -- thanks for raising this @spencerahill.  Indeed I was a little careless about introducing this breaking change.  

> Did the generalizing you've done make that no longer appropriate?

In a way, yes.  My rationale for removing the defaults for `width` and `aspect` was that with the new API, which now includes a `height` argument, it would be awkward, e.g., when creating a height-and-width constrained figure, to have to explicitly set the `aspect` argument to `None`.

There's definitely a trade-off though with convenience.  I agree it was nice in the previous version that reasonable defaults were provided.  I would be open to changing the API if necessary.  Maybe the whole "pick only two out of the `width`, `height`,  and `aspect` arguments" is a bit too magical?  Maybe there should be different functions for different types of figures?  What do you think?

**Comment by spencerahill** [comment]
>it would be awkward, e.g., when creating a height-and-width constrained figure, to have to explicitly set the aspect argument to None

Ah, good point.

>There's definitely a trade-off though with convenience. I agree it was nice in the previous version that reasonable defaults were provided. I would be open to changing the API if necessary. Maybe the whole "pick only two out of the width, height, and aspect arguments" is a bit too magical? Maybe there should be different functions for different types of figures? What do you think?

I don't feel strongly about this.  And I'm not sure that it's too magical.  Convenience is good, but not the only good.  With my wrappers above I've got drop-ins that suit my needs indefinitely.  So I would hesitate to push you toward changing the API (again) based on an n=1 opinion sample.

Thanks again for faceted!  It's great.

**Comment by chuaxr** [comment]
Hi @spencerkclark! Like @spencerahill, I noticed this change after recently creating a new conda environment and would like to add my support for being able to call faceted without having to explicitly specify width/aspect/height information. 

Another possibility would be to supply defaults if (and only if) there are less than two of the width, height, and aspect arguments. `plt.figure` provides a default width and height so having defaults doesn't strike me as being 'too magical'.  

@spencerahill, if you don't mind, can you elaborate on how you're implementing the wrappers? I'm sure there's a better way then copy and pasting those definitions into every notebook/script with plots -- do you install them into a separate personal package as part of your environment?

**Comment by spencerkclark** [comment]
> Another possibility would be to supply defaults if (and only if) there are less than two of the width, height, and aspect arguments. `plt.figure` provides a default width and height so having defaults doesn't strike me as being 'too magical'.

Ah that is clever!  If I'm understanding correctly, I would propose the following behavior:
- If `aspect=None` and `height=None` and a `width` is specified, we would set `aspect=0.618`.
- If `width=None` and `height=None` and an `aspect` is specified, we would set `width=8.0`.
- If `aspect=None` and `width=None` and a `height` is specified, we would set `aspect=0.618`.
- If `aspect=None`, `width=None`, and `height=None`, we would set `aspect=0.618` and `width=8.0`.

I think this would be a way to recover the old defaults without needing to change the current signature of the function.

**Comment by spencerahill** [comment]
Nice hearing from you @chuaxr!  I share @spencerkclark 's enthusiasm about your proposal.

Meantime, here's how I've coded up my wrappers: https://github.com/spencerahill/puffins/blob/master/puffins/plotting.py; see the `def faceted` and `def faceted_ax` lines.

I install the overarching puffins package in each environment I setup via a local install, `pip install -e .` from the puffins directory.  So then I can do e.g. `from puffins.plotting import faceted, faceted_ax` from any script or notebook.

**Comment by chuaxr** [comment]
Thanks both!  (And nice package name too,@spencerahill ;) ) Very much appreciate @spencerkclark sharing faceted as well. It really takes a lot of the trial and error out of making the plot elements fit within the figure without compromising on text size.

**Comment by spencerkclark** [comment]
I implemented @chuaxr's suggestion in #39 and issued a v0.2.1 release.  It should already be available on PyPI!
